src/SDK/Trace/Span.php

```diff
@@ -249,6 +251,15 @@ public function end(int $endEpochNanos = null): void
             return;
         }
 
+        $spanData = $this->toSpanData();
```

Comment:
Would it perhaps make sense to skip logging for not sampled spans? All built-in processor implementations discard them anyway. Another option could be having this whole check in the processors since there's no additional `toSpanData` overhead then.

Reply:
I wasn't really happy with needing the extra `toSpanData` call. Perhaps we could move the check into `toSpanData` itself (which would not be called on a non-sampled span). That effectively moves the check into the processors, without duplicating the code into each sampler? (although, it could be in a trait to remove that duplication)...